api-metastore/src/main/java/org/zalando/nakadi/service/SubscriptionService.java

```diff
@@ -153,6 +164,29 @@ public Subscription createSubscription(final SubscriptionBase subscriptionBase)
         return subscription;
     }
 
+    private Subscription createSubscriptionWithEventTypeLock(final SubscriptionBase subscriptionBase) {
+        try {
+            return transactionTemplate.execute(action -> {
+                final Set<String> dbEventTypes = eventTypeRepository.
+                        listEventTypesWithRowLock(subscriptionBase.getEventTypes()).stream().
+                        map(EventType::getName).collect(Collectors.toSet());
+
+                if(dbEventTypes.size() != subscriptionBase.getEventTypes().size() ){
+                    final List<String> missingEventTypes = subscriptionBase.getEventTypes().stream().
+                            filter(name -> !dbEventTypes.contains(name)).collect(Collectors.toList());
+
+                    throw new NoSuchEventTypeException(String.format("Failed to create subscription, " +
+                            "event type(s) not found: '%s'", String.join("', '", missingEventTypes)));
+                }
+
+                return subscriptionRepository.createSubscription(subscriptionBase);
+            });
+        } catch (TransactionException e) {
+            LOGGER.error("Failed to create subscription, unable to obtain lock", e);
+            throw new InconsistentStateException("Failed to create subscription, unable to obtain lock");
```

Comment:
Let's log or re-throw, but not both: the exception should be logged where it's caught ultimately.

Also not sure that "unable to obtain lock" is the only reason for `TransactionException` — should we really mention that?